File: clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppListener.java
package com.clevertap.android.sdk.inapp;

import android.content.Context;
import android.os.Bundle;
import java.util.HashMap;

public interface InAppListener {

    void inAppNotificationDidClick(CTInAppNotification inAppNotification, Bundle formData,
            HashMap<String, String> keyValueMap);

    void inAppNotificationDidDismiss(Context context, CTInAppNotification inAppNotification, Bundle formData);

    void inAppNotificationDidShow(CTInAppNotification inAppNotification, Bundle formData);
}

File: clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppListener.kt
package com.clevertap.android.sdk.inapp

import android.content.Context
import android.os.Bundle

internal interface InAppListener {

    fun inAppNotificationDidClick(
        inAppNotification: CTInAppNotification,
        button: CTInAppNotificationButton,
        activityContext: Context?
    ): Bundle?

    fun inAppNotificationDidDismiss(inAppNotification: CTInAppNotification, formData: Bundle?)
    fun inAppNotificationDidShow(inAppNotification: CTInAppNotification, formData: Bundle?)
    fun inAppNotificationActionTriggered(
        inAppNotification: CTInAppNotification,
        action: CTInAppAction,
        callToAction: String,
        additionalData: Bundle?,
        activityContext: Context?
    ): Bundle?
}
